File: src/main/java/com/eager/questioncloud/workspace/WorkSpaceQuestionRemover.java
package com.eager.questioncloud.workspace;

import com.eager.questioncloud.question.Question;
import com.eager.questioncloud.question.QuestionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class WorkSpaceQuestionRemover {
    private final QuestionRepository questionRepository;

    public void remove(Long creatorId, Long questionId) {
        Question question = questionRepository.getForModifyAndDelete(questionId, creatorId);
        question.delete();
        questionRepository.save(question);
    }
}

File: qc-core/qc-creator/qc-creator-api/src/main/kotlin/com/eager/questioncloud/workspace/implement/WorkspaceQuestionRemover.kt
package com.eager.questioncloud.workspace.implement

import com.eager.questioncloud.common.exception.CoreException
import com.eager.questioncloud.common.exception.Error
import com.eager.questioncloud.creator.repository.CreatorRepository
import com.eager.questioncloud.question.api.internal.QuestionCommandAPI
import com.eager.questioncloud.workspace.command.DeleteQuestionCommand
import org.springframework.stereotype.Component

@Component
class WorkspaceQuestionRemover(
    private val creatorRepository: CreatorRepository,
    private val questionCommandAPI: QuestionCommandAPI,
) {
    fun deleteQuestion(command: DeleteQuestionCommand) {
        val creator = creatorRepository.findByUserId(command.userId) ?: throw CoreException(Error.NOT_FOUND)
        questionCommandAPI.delete(command.questionId, creator.id)
    }
}